wolfcrypt/src/asn.c

```diff
@@ -741,6 +741,128 @@ static word32 SetBitString(word32 len, byte unusedBits, byte* output)
 }
 #endif /* !NO_RSA || HAVE_ECC || HAVE_ED25519 */
 
+#ifdef ASN_BER_TO_DER
+/* Convert a BER encoding with indefinite length items to DER.
+ *
+ * ber     BER encoded data.
+ * length  Length of BER encoded data.
+ * der     Buffer to hold DER encoded version of data.
+ *         NULL indicates only the length is required.
+ * returns the length of the DER data on success, ASN_PARSE_E if the BER data is
+ *         invalid and BAD_FUNC_ARG if ber is NULL.
+ */
+int wc_BerToDer(byte* ber, word32 length, byte* der)
```

Comment:
Since this is a public function we should have a length input for Der size as well so we don't overwrite the Der buffer.  It will also catch programming errors on our side.